### PR TITLE
Upgrade slack v3

### LINKED ISSUE #3: Upgrade to slack-sdk from slackclient
Slack, in their infinite wisdom, have deprecated `slackclient` and the RTM connection method. It will still continue to work, but we'll need to update sooner or later to the new method. They at least give us socket mode now, which should help ease some of the pain.

* migration docs: https://slack.dev/python-slack-sdk/v3-migration/
* socket mode: https://api.slack.com/apis/connections/socket

This change is needed before I can finish building the "recover from faulty deploy" mode that I'm working on, as we can't kill the RTMClient without some sort of external input once it's started.

### PR DESCRIPTION
**Pull request by itsthejoker**
Closes #3 

## Description:

Since Slack wants us to move to the events architecture and I want to add autorecovery to remote deploys, here we are. Everything works except the presence system (checking to see if people are online), which we can deploy without because it's an optional system. Deploying this will require changes to the server (namely just installing the new dependencies) but will require more if we want to rename `bubblesRTM.py` since... well, it's not an RTM-based bot anymore.

## Checklist:

- [x] Code Quality
- [x] Pep-8
- [ ] Tests (if applicable)
- [x] Success Criteria Met
- [x] Inline Documentation
- [ ] Wiki Documentation (if applicable)

### CODE REVIEW

**Review by Pf441 — APPROVED** [review]
I see no major issues with that, so I guess it's good.

File: bubbles/commands/periodic/activity_checkin.py [hunk 1]
```diff
@@ -79,18 +71,18 @@ def presence_update_callback(*args, **kwargs):
...
-    # to force a refresh of everyone and make sure that they're actually offline.
-    # Instead of forcing a refresh when we check (which would require some dancing
-    # with slack to get the timing right) we just schedule the force update as its
-    # own thing and carry on.
-    # This function is only responsible for triggering the request; the result is
-    # returned via a RTM event captured in bubblesRTM.py.
-    rtm_client.send_over_websocket(
-        payload={"type": "presence_query", "ids": USER_IDS,}
-    )
+# def force_presence_update(rtm_client: RTMClient):
```
Comment (MaxVanDeursen) [inline bubbles/commands/periodic/activity_checkin.py:75]: We should probably relate this snippet of commented code to that above.

File: bubbles/config.py [hunk 2]
```diff
@@ -28,26 +28,30 @@
...
+    user_agent=os.environ.get("reddit_user_agent"),
 )
 
-client = WebClient(token=API_KEY)
-rtm_client = RTMClient(token=API_KEY)
+app = App(
+    signing_secret=os.environ.get("slack_signing_secret"),
+    token=os.environ.get("slack_oauth_token"),
+)
+# client = WebClient(token=API_KEY)
```
Comment (MaxVanDeursen) [inline bubbles/config.py]: We can probably remove this commented code?

File: bubblesRTM.py [hunk 3]
```diff
@@ -1,56 +1,97 @@
...
+def message_received(ack, payload):
+    ack()
     try:
-        process_message(**payload)
-    except Exception:
-        client.chat_postMessage(
-            channel=payload["data"]["channel"],
+        process_message(payload)
+    except:
+        print("AAAACK")
```
Comment (MaxVanDeursen) [inline bubblesRTM.py]: Remove print message?

**Review by MaxVanDeursen — CHANGES_REQUESTED** [review]
Overall, a really nice and tidy PR! I left some comments regarding the commented code, but other than that it looks great!

File: bubblesRTM.py (hunk 3 above)
Comment (itsthejoker) [reply bubblesRTM.py]: ooooooooooops

**Pull request merged**